ext/standard/basic_functions.c

```diff
@@ -1969,6 +1969,20 @@ PHP_FUNCTION(highlight_string)
 }
 /* }}} */
 
+/* {{{ Get a byte size from the ini byte size shorthand */
+PHP_FUNCTION(ini_bytes)
+{
+	char *shorthand;
+	size_t shorthand_len;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_STRING(shorthand, shorthand_len)
+	ZEND_PARSE_PARAMETERS_END();
+
+	RETVAL_LONG(zend_atol(shorthand, shorthand_len));
```

Comment:
Can it fail? What happens if it fails? I see no test for valid and non-valid values.

Reply:
I don't believe that the function can fail and there are no truly invalid values. That is, for values that aren't valid from a use standpoint they will return `0`.

For certain parameters like `post_max_size` that means if we don't have a legitimate configuration value we won't apply any limit, whereas a legitimate value is any value that returns > 0 from this function, but negative values and outright ridiculous values are still allowed.